compiler/luci/pass/src/DecomposeHardSwishPass.cpp

```diff
@@ -0,0 +1,147 @@
+ *              |
+ *        [CircleNode]
+ *
+ *
+ *  AFTER
+ *
+ *      [CircleNode]  [CircleConst]
+ *          |    \       /
+ *          |     \     /
+ *          |   [CircleAdd]
+ *          |        |
+ *          |        |
+ *          \  [CircleRelu6] [CircleConst]
+ *           \        \        /
+ *            \        \      /
+ *             \      [CircleMul]
+ *              \       /
+ *               \     /
+ *             [CircleMul]
+ *                  |
+ *                  |
+ *             [CircleNode]
+ *
+ */
+template <loco::DataType DT> bool decompose_hardswish(luci::CircleHardSwish *hardswish)
+{
+  if (not hardswish)
+    return false;
+
+  if (hardswish->dtype() != DT)
+    return false;
+
+  auto g = hardswish->graph();
+
+  auto name = hardswish->name();
+  assert(name.length() > 0);
+
+  // Create a const for CircleAdd operation
+  auto add_const = g->nodes()->create<luci::CircleConst>();
+  add_const->shape({}); // scalar
+  add_const->dtype(DT);
+  add_const->rank(0);
+  add_const->size<DT>(1);
+  add_const->at<DT>(0) = 3.;
+  add_const->name("add_const");
+  luci::add_origin(add_const, luci::get_origin(hardswish));
+
+  // Create an Add operation
+  auto add = g->nodes()->create<luci::CircleAdd>();
+  add->fusedActivationFunction(luci::FusedActFunc::NONE); // TODO: Seungho
+  add->x(hardswish->features());
+  add->y(add_const);
+  add->name(name + "/Add");
+  luci::add_origin(add, luci::get_origin(hardswish));
+
+  // Create a Relu6 operation
+  auto relu6 = g->nodes()->create<luci::CircleRelu6>();
+  relu6->features(add);
+  relu6->name(name + "/Relu6");
+  luci::add_origin(relu6, luci::get_origin(hardswish));
+
+  // Create a const for CircleMul operation
+  auto mul_const = g->nodes()->create<luci::CircleConst>();
+  mul_const->shape({}); // scalar
+  mul_const->dtype(DT);
+  mul_const->rank(0);
+  mul_const->size<DT>(1);
+  mul_const->at<DT>(0) = 1. / 6.;
+  mul_const->name("mul_const");
+  luci::add_origin(mul_const, luci::get_origin(hardswish));
+
+  // Create first Mul operation
+  auto mul1 = g->nodes()->create<luci::CircleMul>();
+  mul1->fusedActivationFunction(luci::FusedActFunc::NONE);
+  mul1->x(relu6);
+  mul1->y(mul_const);
+  mul1->name(name + "/Mul1");
+  luci::add_origin(mul1, luci::get_origin(hardswish));
+
+  // Create second Mul operation
+  auto mul2 = g->nodes()->create<luci::CircleMul>();
+  mul2->fusedActivationFunction(luci::FusedActFunc::NONE);
+  mul2->x(hardswish->features());
+  mul2->y(mul1);
+  mul2->name(name + "/Mul2");
+  luci::add_origin(mul2, luci::get_origin(hardswish));
+
+  replace(hardswish).with(mul2);
+
+  return true;
+}
+
+} // namespace
+
+namespace luci
+{
+
+bool DecomposeHardSwishPass::run(loco::Graph *g)
+{
+  bool changed = false;
+
+  for (auto node : loco::active_nodes(loco::output_nodes(g)))
+  {
+    if (auto hardswish = dynamic_cast<luci::CircleHardSwish *>(node))
+    {
+      if (decompose_hardswish<loco::DataType::FLOAT32>(hardswish))
```

Comment:
Q) Do you have plan to support other types?

Reply:
> mul_const->at<DT>(0) = 1. / 6.;

this line my prevent to support int types.

Reply:
Currently, I am only considering the FLOAT32 format.
I think it is better to remove template now.

Reply:
Applied =)